packages/istanbul-lib-report/lib/context.js

```diff
@@ -41,12 +42,79 @@ function mergeWatermarks(specified, defaults) {
  *  filesystem lookups based on path.
  * @constructor
  */
-function Context(opts) {
-    opts = opts || {};
-    this.dir = opts.dir || 'coverage';
-    this.watermarks = mergeWatermarks(opts.watermarks, watermarks.getDefault());
-    this.sourceFinder = opts.sourceFinder || defaultSourceLookup;
-    this.data = {};
+class Context {
+    constructor(opts) {
+        opts = opts || {};
+        this.dir = opts.dir || 'coverage';
+        this.watermarks = mergeWatermarks(
+            opts.watermarks,
+            watermarks.getDefault()
```

Comment:
can `mergeWatermarks` make this an optional argument? 

on second thought that might not be clearer...

Reply:
Possibly worth doing in a follow-up.